tests/transforms/test_split_non_commuting.py

```diff
@@ -953,3 +955,93 @@ def test_trainable_hamiltonian_tensorflow(self, grouping_strategy):
         actual = tape.jacobian(cost, params)
 
         assert qml.math.allclose(actual, [-0.5, np.cos(np.pi / 4)])
+
+
+# Single hamiltonian expval measurement to test shot distribution
+ham = qml.Hamiltonian(
+    coeffs=[10, 0.1, 20, 100, 0.2],
+    observables=[
+        qml.X(0) @ qml.Y(1),
+        qml.Z(0) @ qml.Z(2),
+        qml.Y(1),
+        qml.X(1) @ qml.X(2),
+        qml.Z(0) @ qml.Z(1) @ qml.Z(2),
+    ],
+)
+ham_expval = qml.expval(ham)
+total_shots = 1000
+coeffs_per_group = [[10, 20], [0.1, 0.2], [100]]
+
+
+class TestShotDistribution:
+    """
+    Test shot distribution for the `split_non_commuting` transform.
+    At the moment, this feature is available only for the single hamiltonian
+    measurement case and with "default" or "qwc" grouping_strategy.
+    """
+
+    @pytest.mark.parametrize("grouping_strategy", ["default", "qwc", "wires", None])
+    def test_none_shot_distribution(self, grouping_strategy):
+        """Test standard behaviour with no shot distribution strategy."""
+
+        initial_tape = qml.tape.QuantumScript(measurements=[ham_expval], shots=total_shots)
+
+        tapes, _ = split_non_commuting(
+            initial_tape,
+            grouping_strategy=grouping_strategy,
+            shot_distribution=None,
+        )
+
+        assert sum(tape.shots.total_shots for tape in tapes) == total_shots * len(tapes)
+
+        for tape in tapes:
+            assert tape.shots.total_shots == total_shots
+
+    @pytest.mark.parametrize("grouping_strategy", ["default", "qwc"])
+    @pytest.mark.parametrize(
+        ["shot_distribution", "expected_shots"],
+        [
+            ("uniform", (334, 333, 333)),
+            ("weighted", (231, 2, 767)),
+            ("weighted_random", (221, 3, 776)),  # requires fixed seed=42
```

Comment:
Though seeding is deterministic on a per-setup basis, it isn't necessarily going to be the same between versions and computers. So then the question is about how to we actually test this?  You will probably get different answers from different people, but I would probably test things like:

1) It's not exactly weighted, but sufficiently closed to weighted.
2) The seed changes the exact values

Reply:
Mh ok yes, that' right...

> 1. It's not exactly weighted, but sufficiently closed to weighted.
> 2. The seed changes the exact values

Do you mean something like checking that the result for `weighted_random` is close to `weighted` up to some tolerance (e.g. 10 or 20%)?

Reply:
@albi3ro @JerryChen97 Maybe something like [db7fe30](https://github.com/PennyLaneAI/pennylane/pull/7988/commits/db7fe30adab8254577814d10fdd125228842801a)?